internal/pipe/ko/ko.go

```diff
@@ -0,0 +1,269 @@
+package ko
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"golang.org/x/tools/go/packages"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/ko/pkg/build"
+	"github.com/google/ko/pkg/commands/options"
+	"github.com/google/ko/pkg/publish"
+	"github.com/goreleaser/goreleaser/internal/ids"
+	"github.com/goreleaser/goreleaser/internal/semerrgroup"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	gcontext "github.com/goreleaser/goreleaser/pkg/context"
+)
+
+// Pipe that catalogs common artifacts as an SBOM.
+type Pipe struct{}
+
+func (Pipe) String() string { return "running ko" }
+func (Pipe) Skip(ctx *gcontext.Context) bool {
+	return ctx.SkipKo || ctx.Config.Ko.ID == ""
+}
+
+// Default sets the Pipes defaults.
+func (Pipe) Default(ctx *gcontext.Context) error {
+	ids := ids.New("kos")
+	cfg := &ctx.Config.Ko
+	if err := setConfigDefaults(cfg); err != nil {
+		return err
+	}
+	ids.Inc(cfg.ID)
+	return ids.Validate()
+}
+
+func setConfigDefaults(cfg *config.Ko) error {
+	cfg.Push = true
+
+	if cfg.ID == "" {
+		cfg.ID = "default"
+	}
+
+	if cfg.BaseImage == "" {
+		cfg.BaseImage = "cgr.dev/chainguard/static" // TODO: we can discuss on this
+	}
+
+	return nil
```

Comment:
do we want to add the same default ldflags we do in the goreleaser build here as well?